### PR TITLE
Draw all horizons and sky base, in front of stars

### LINKED ISSUE #7920: Stars are rendered below horizon
screenshot_20181201_042216

Quite often the player is high up and much of the terrain below is not loaded.

### PR DESCRIPTION
**Pull request by paramat**
Draw all horizons and sky base, in front of stars

Move star draw to before sun glow texture draw and before sun draw,
not currently essential but the logical order. Will be necessary if
a 'no far ground' option is added, to draw stars behind the sun.
/////////////////////////

screenshot_20181203_052922

^ PR

Closes #7920 
When the player is high enough for ground level to not be rendered or loaded, instead of seeing stars through the MT world below there is now a sense of the dark world below.

Recent testing in MT master with more and larger stars shows that stars are already hidden below E and W horizons:

screenshot_20181202_062332

Now we know of this, the dark rectangle is noticeable on the left with normal stars in MT master:

screenshot_20181201_042216

As the stars rotate with time they appear from behind, and disappear behind, these E and W rectangles.
This PR fixes the inconsistency, completes the horizons and moves the sky base drawing to later to be in front of sun, moon and stars. Only 2 more quads are drawn.
/////////////////////////////

Additionally we could have a 'groundless' option that does not draw the 'far cloudy fog things', the sun horizon glow texture, or the 5 quads that hide sun, moon and stars.
In this case the sun and moon would have to always be drawn for their complete circles. This would be like being in deep space between sun and moon, in daytime the blue sky would suggest a completely air world hanging in space.
This would be suitable for 'groundless' worlds like skyblock or floating islands, and if time is set to night, suitable for space games.

I could code this later if it's decided to be necessary.

### CODE REVIEW

**Comment by paramat** [comment]
Indeed, i can add a clientside (for now, could possibly be part of 'set sky' later) option for that as described if required, either in this PR or in a follow-up PR.

**Comment by paramat** [comment]
2nd commit added to add 'groundless' option, first post updated.

**Comment by Glory7000** [comment]
I would approve commit 1, it's a nice touch, good job, but I have not contributed to Minetest. I think everyone will also agree to merge it.

However, I think commit 2 can be controversial because there are games that don't need it and others that do, and that was smart of you to request comments on the forum. *I think* it should be merged as well as it will make space themed or surreal themed games easier, such as Skyblock-clones and space adventures. Even for end users this has a positive effect, those just wanting to build a spaceship in singlenode. Obviously, it's not server-side but nonetheless would be an improvement over the current horizon space hybrid we've got right now IIRC.

Cheers.

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -796,6 +796,13 @@ autoscale_mode (Autoscaling mode) enum disable disable,enable,force
 #    Show entity selection boxes
 show_entity_selectionbox (Show entity selection boxes) bool true
 
+#    If 'true', the skybox displays horizon haze, sunrise/set horizon glow and
+#    terrain horizons that hide the sun, moon and stars.
+#    Set to 'false' to create the appearence of space (if time is set to night)
+#    or a 'terrainless' air world (if time is set to day).
+#    If 'false', 'directional colored fog' is not effective so is disabled.
+skybox_far_terrain (Show skybox far terrain) bool true
```
Comment (rubenwardy) [inline builtin/settingtypes.txt]: This should be a hud flag, really, as it's a game play related thing

**Comment by paramat** [comment]
I'm collecting opinions in the forum https://forum.minetest.net/viewtopic.php?f=5&t=21528
I think i prefer to merge commit 1 now as it's obvious, can always code something later.

File: builtin/settingtypes.txt (hunk 1 above)
Comment (paramat) [reply builtin/settingtypes.txt]: 2nd commit removed.

**Pull request merged**